scripts/updates/auto-updates.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+set -e -u
+BASEDIR=$(dirname "$(realpath "$0")")
+PKGSDIR="${BASEDIR}/../../packages"
+
+# These variables should be set in environment outside of this script.
+# Build updated packages.
+: "${BUILD_PACKAGES:=false}"
+# Commit changes to Git.
+: "${GIT_COMMIT_PACKAGES:=false}"
+# Push changes to remote.
+: "${GIT_PUSH_PACKAGES:=false}"
+
+if [ -z "${GITHUB_API_TOKEN-}" ]; then
+	echo "You need a Github Personal Access Token be set in variable GITHUB_API_TOKEN."
+	exit 1
+fi
+
+for package in $(ls -1 "${PKGSDIR}"); do
+	unset auto_update_flag project version_regexp termux_version termux_epoch latest_version srcurl
+
+	# Extract some build vars
+	build_vars=$(set +e +u; . "${PKGSDIR}/${package}/build.sh" 2>/dev/null; \
+		echo "auto_update_flag=${TERMUX_PKG_AUTO_UPDATE}"; \
+		echo "termux_version=${TERMUX_PKG_VERSION}"; \
+		echo "srcurl=${TERMUX_PKG_SRCURL}" ;\
+	)
+
+	# Eval build vars to local variables
+	source <(echo "$build_vars")
+
+	if [ "${auto_update_flag}" != "true" ]; then
+		echo "[SKIP] ${package} is not auto updatable."
+		continue
+	fi
+	
+	termux_epoch="$(echo "$termux_version" | cut -d: -f1)"
+	termux_version=$(echo "$termux_version" | cut -d: -f2-)
+	if [ "$termux_version" == "$termux_epoch" ]; then
+		# No epoch set.
+		termux_epoch=""
+	else
+		termux_epoch+=":"
+	fi
+
+	# Transform an actual version to regexp
+	version_regexp=$(echo "${termux_version}" | sed s@[0-9]@\\\\d+@g)
+
+	# Extract github project from TERMUX_PKG_SRCURL
+	project="$(echo "${srcurl}" | grep github.com | cut -d / -f4-5)"
+	if [ -z "${project}" ]; then
+		echo "[SKIP] ${package}'s TERMUX_PKG_SRCURL is not a github archive url."
+		continue
```

Comment:
Here must be an error. If package was configured for auto updates by mistake, we should get notification that auto updates have a problem.